packages/wonder-blocks-core/util/types.js

```diff
@@ -132,13 +132,92 @@ export type AriaProps = {
     "aria-valuemin"?: number,
     "aria-valuenow"?: number,
     "aria-valuetext"?: string,
-};
+|};
 
-export type Props = AriaProps & {
-    style?: StyleType,
+type MouseEvents = {|
+    onMouseDown?: (e: SyntheticMouseEvent<*>) => void,
+    onMouseUp?: (e: SyntheticMouseEvent<*>) => void,
+    onMouseMove?: (e: SyntheticMouseEvent<*>) => void,
+    onClick?: (e: SyntheticMouseEvent<*>) => void,
+    onDoubleClick?: (e: SyntheticMouseEvent<*>) => void,
+    onMouseEnter?: (e: SyntheticMouseEvent<*>) => void,
+    onMouseLeave?: (e: SyntheticMouseEvent<*>) => void,
+    onMouseOut?: (e: SyntheticMouseEvent<*>) => void,
+    onMouseOver?: (e: SyntheticMouseEvent<*>) => void,
+    onDrag?: (e: SyntheticMouseEvent<*>) => void,
+    onDragEnd?: (e: SyntheticMouseEvent<*>) => void,
+    onDragEnter?: (e: SyntheticMouseEvent<*>) => void,
+    onDragExit?: (e: SyntheticMouseEvent<*>) => void,
+    onDragLeave?: (e: SyntheticMouseEvent<*>) => void,
+    onDragOver?: (e: SyntheticMouseEvent<*>) => void,
+    onDragStart?: (e: SyntheticMouseEvent<*>) => void,
+    onDrop?: (e: SyntheticMouseEvent<*>) => void,
+|};
+
+type KeyboardEvents = {|
+    onKeyDown?: (e: SyntheticKeyboardEvent<*>) => void,
+    onKeyPress?: (e: SyntheticKeyboardEvent<*>) => void,
+    onKeyUp?: (e: SyntheticKeyboardEvent<*>) => void,
+|};
+
+type InputEvents = {|
+    onChange?: (e: SyntheticInputEvent<*>) => void,
+    onInput?: (e: SyntheticInputEvent<*>) => void,
+    onInvalid?: (e: SyntheticInputEvent<*>) => void,
+    onSubmit?: (e: SyntheticInputEvent<*>) => void,
+|};
+
+type TouchEvents = {|
+    onTouchCancel?: (e: SyntheticTouchEvent<*>) => void,
+    onTouchEnd?: (e: SyntheticTouchEvent<*>) => void,
+    onTouchMove?: (e: SyntheticTouchEvent<*>) => void,
+    onTouchStart?: (e: SyntheticTouchEvent<*>) => void,
+|};
+
+type FocusEvents = {|
+    onFocus?: (e: SyntheticFocusEvent<*>) => void,
+    onBlur?: (e: SyntheticFocusEvent<*>) => void,
+|};
+
+type EventHandlers = {|
+    ...MouseEvents,
+    ...KeyboardEvents,
+    ...InputEvents,
+    ...TouchEvents,
+    ...FocusEvents,
+    // TODO: add remaining supported events https://reactjs.org/docs/events.html#supported-events
+|};
+
+export type CommonProps = {|
```

Comment:
I should maybe renamed this `SharedProps` since we use that in other files.

Reply:
I think that either way, it's not a great name since it doesn't really say what it is for; (what is it common or shared with?). I think I would like to see a name that conveyed when it should be used rather than that it is shared in any way. In lieu of that (or perhaps in addition to that), it needs some commentary to explain its purpose.